### PR TITLE
Introduction FileInfo.Lock() to lock files during a test

### PR DESCRIPTION
**Pull request by Corniel**
To test IO related unhappy flows, it can be useful to create a temporary lock on a file. This can be done as follows:

``` C#

var file = new FileInfo("somefile.txt");

using (var @lock = file.Lock())
{
     file.OpenRead(); // Throws IOException.
}
```

### CODE REVIEW

File: specs/Qowaiv.Specs/Clock_specs.cs [hunk 1]
```diff
@@ -103,24 +103,6 @@ public void UtcNow_and_TimeZone_can_be_set_together()
         Clock.UtcNow().Should().NotBe(Svo.DateTime);
         Clock.TimeZone.Should().NotBe(Svo.TimeZone);
     }
-
```
Comment (Sjaaky) [inline specs/Qowaiv.Specs/Clock_specs.cs:106]: Seems unrelated

Comment (Corniel) [reply specs/Qowaiv.Specs/Clock_specs.cs:106]: Was a flacky test. But, yes unrelated.
